src/graph/Graph.js

```diff
@@ -501,10 +501,20 @@
                 context._selection.click(d, d3.event);
             })
             .on("click", function (d) {
-                context.vertex_click(d, d3.event);
+                var vertexElement = d3.select(this).select(".graph_Vertex");
+                var selected = false;
+                if (!vertexElement.empty()) {
+                    selected = vertexElement.classed("selected");
+                }
+                context.vertex_click(d.data(), "", selected);
```

Comment:
@jbrundage I think I should change this to:
```
context.vertex_click(d.data(), "", selected, {
   graphVertex:  d 
});
```
You will still need a refactor, but it should be a small one?